DetectorDescription/DDCMS/plugins/DDCompactViewESProducer.cc

```diff
@@ -48,8 +48,7 @@ class DDCompactViewESProducer : public edm::ESProducer {
 };
 
 DDCompactViewESProducer::DDCompactViewESProducer(const edm::ParameterSet& iConfig)
-    : m_detToken(setWhatProduced(this).consumes<DDDetector>(
-          edm::ESInputTag("", iConfig.getParameter<std::string>("appendToDataLabel")))) {}
+    : m_detToken(setWhatProduced(this).consumes()) {}
```

Comment:
@makortel I think this change that you recommended has to be reverted. There are a number of configs that use the `appendToDataLabel`, like:
https://cmssdt.cern.ch/lxr/source/SimTracker/TrackerMaterialAnalysis/test/listGroups_dd4hep.py

and they fail without `appendToDataLabel`. Is it safe to leave leave `DDCompactViewESProducer.cc` in its original form?

Reply:
I'd highly recommend to change the configurations that use these ESProduders with `appendToDataLabel` for the data label for their input products instead. As discussed above that pattern causes problems.

Reply:
Written that, you could add a new parameter to that allows to specify the data label of the input product, and continue to use the `appendToDataLabel` to set the label of the output product, if there are cases where that would be useful (it is not evident to me that the example you pointed to really needs the data labels).

Reply:
@makortel I agree (and initially proposed) that these codes be changed to add a new parameter of type `ESInputTag` probably named `src` to specify the origin of data. That keeps things obvious and gives the most flexibility.

Given that, this case is a bit strange since there are multiple DDCompactViews floating around (designated by "CMS" and "MagField" strings). Those two groups should never be mixed. So for this case, using the `appendToDataLabel` as a way to guarantee consistency may be of value.

The item that really caused a problem was using `@module_label` for the produces call which we then removed.

Reply:
> Given that, this case is a bit strange since there are multiple DDCompactViews floating around (designated by "CMS" and "MagField" strings). Those two groups should never be mixed. So for this case, using the `appendToDataLabel` as a way to guarantee consistency may be of value.

Good point, thanks. I agree, in such a case using the `appendToDataLabel` also for input data label would add guarantees (and be simpler).

Reply:
@Dr15Jones @makortel Sorry, I'm not following. Are you saying I should leave this file in its original state, with possibly the addition of a new `src` parameter? I would think the new parameter could be added in a separate PR.

Reply:
@cvuosalo we agree to revert this file and that no additional later pull request are needed (unless something else turns up).